include/stdx/atomic.hpp

```diff
@@ -0,0 +1,126 @@
+                  "Atomic values must be trivially copyable, copy "
+                  "constructible and copy assignable");
+
+    using elem_t = ::atomic::atomic_type_t<T>;
+    constexpr static auto alignment = ::atomic::alignment_of<T>;
+
+    static_assert(std::is_convertible_v<elem_t, T>,
+                  "::atomic::atomic_type is specialized, but the type it "
+                  "produces is not convertible to T");
+    static_assert(std::is_convertible_v<T, elem_t>,
+                  "::atomic::atomic_type is specialized, but the type it "
+                  "produces is not convertible from T");
+
+    alignas(alignment) elem_t value;
+
+  public:
+    using value_type = T;
+
+    constexpr atomic()
+#if __cplusplus >= 202002L
+        requires std::is_default_constructible_v<elem_t>
+#endif
+        : value{} {
+    }
+    constexpr atomic(T t) : value{static_cast<elem_t>(t)} {}
+    atomic(atomic const &) = delete;
+    auto operator=(atomic const &) -> atomic & = delete;
+
+    [[nodiscard]] auto
+    load(std::memory_order mo = std::memory_order_seq_cst) const -> T {
+        return static_cast<T>(::atomic::load(value, mo));
+    }
+
+    void store(T t, std::memory_order mo = std::memory_order_seq_cst) {
+        ::atomic::store(value, static_cast<elem_t>(t), mo);
+    }
+
+    operator T() const { return load(); }
+    auto operator=(T t) -> T {
+        store(t);
+        return t;
+    }
+
+    [[nodiscard]] auto
+    exchange(T t, std::memory_order mo = std::memory_order_seq_cst) -> T {
+        return ::atomic::exchange(value, static_cast<elem_t>(t), mo);
+    }
+
+    auto fetch_add(T t, std::memory_order mo = std::memory_order_seq_cst) -> T {
+        CPP20(static_assert(
+            requires { t + t; }, "operation is not supported on T"));
+        return ::atomic::fetch_add(value, static_cast<elem_t>(t), mo);
+    }
+    auto fetch_sub(T t, std::memory_order mo = std::memory_order_seq_cst) -> T {
+        CPP20(static_assert(
+            requires { t - t; }, "operation is not supported on T"));
+        return ::atomic::fetch_sub(value, static_cast<elem_t>(t), mo);
+    }
+
+    auto operator+=(T t) -> T { return fetch_add(t) + t; }
+    auto operator-=(T t) -> T { return fetch_sub(t) - t; }
+
+    auto operator++() -> T {
+        CPP20(static_assert(
+            requires(T t) { ++t; }, "Operation is not supported on T"));
+        return fetch_add(1) + 1;
+    }
+    [[nodiscard]] auto operator++(int) -> T {
+        CPP20(static_assert(
+            requires(T t) { t++; }, "Operation is not supported on T"));
+        return fetch_add(1);
+    }
+    auto operator--() -> T {
+        CPP20(static_assert(
+            requires(T t) { --t; }, "Operation is not supported on T"));
+        return fetch_sub(1) - 1;
+    }
+    [[nodiscard]] auto operator--(int) -> T {
+        CPP20(static_assert(
+            requires(T t) { t--; }, "Operation is not supported on T"));
+        return fetch_sub(1);
+    }
```

Comment:
This might be a few questions:
a) Inc/dec are a different operations from operator+/- and it's possible they are supported but not generalised add/sub.
b) It makes no sense to do inc/dec on atomic<T> if it's not supported by T. e.g. if `atomic<bool>` actually becomes `atomic<int>` because of platform alignment/size reqs, it would be possible to do operator-- on it, but it's not something the API should support.

Reply:
Which means actually inc and dec should not be implemented in terms of `fetch_add` and `fetch_sub` here. Done.